functioncall/functioncall.go

```diff
@@ -38,12 +39,21 @@ func (funcs Funcs) MarshalJSON() ([]byte, error) {
 
 func (params Params) MarshalJSON() ([]byte, error) {
 	required := []string{}
-	props := map[string]Param{}
+	props := map[string]any{}
 	for _, p := range params {
 		if p.Required {
 			required = append(required, p.Name)
 		}
-		props[p.Name] = p
+		if p.Type == "array" && p.Items != nil {
```

Comment:
This kind of condition handling should be inside `MarshalJSON` of `Param` itself.

Reply:
func (p Param) MarshalJSON() ([]byte, error)
"We don't need to change this method; using the default is the best."
"If you have changed func (p Param) MarshalJSON() ([]byte, error), then to implement the original functionality you'll need:

```go
var propMap map[string]any
if err := json.Unmarshal(propData, &propMap); err != nil {
	return nil, err
}
return json.Marshal(propMap)
```